app/services/data_api/tad_application_export.rb

```diff
@@ -60,6 +60,9 @@ def as_json
         course_name: application_choice.current_course.name,
         course_code: application_choice.current_course.code,
         nctl_subject: concatenate(application_choice.current_course.subjects.map(&:code)),
+        offer_originally_deferred_at:,
+        offer_reconfirmed_at: application_choice_deferred_confirmed_at,
+        offer_reconfirmed_cycle_year: CycleTimetable.current_year(application_choice_deferred_confirmed_at),
```

Comment:
Oh, I understand now. My preference is that the work on determining whether this value should be nil should be done here, not abstracted to the method above. 

```suggestion
        offer_reconfirmed_cycle_year: application_choice_deferred_confirmed_at.present ? CycleTimetable.current_year(application_choice_deferred_confirmed_at) : nil,
```

Reply:
Yep, makes sense